### DISCUSSION TITLE
Switch status to be shown in Main Menu

### DISCUSSION THREAD

**Discussion by dali-dbr** [arendst/Tasmota#21487] [Q&A] [answered]
Hi.

I'm doing a multi-input door/windows sensor using magnetic sensors.
![image](https://github.com/arendst/Tasmota/assets/6317229/01df19ae-b9d8-4140-a13b-b0c5e5b042c9)

I managed to get 14 GPIO inputs working and sending MQTT. 
![image](https://github.com/arendst/Tasmota/assets/6317229/cb30c1a7-a2f5-4781-bcc1-e9e9bbaec0ef)

Is there any change Switch will show the state in the Main Menu?
![image](https://github.com/arendst/Tasmota/assets/6317229/63659320-f437-440b-b6d2-dbb527975f77)

Thank you

**Comment by sfromis** [comment, marked as answer]
No.

The deliberately rudimentary Tasmtoa web UI is meant for admin/test, not so much as "the place" for daily operation, which is generally expected to be done in your backend setup.

You *could* create a bunch of fake "power" channels which would be linked to the switch input, those would get buttons on the front page. Of course, not useful to click those.

Since you use ESP32, you could do something with Berry programming formatting values, and injecting into the front page.

For testing purposes, the "GPIO Viewer" feature could also be quite useful.

**Reply by dali-dbr** [reply]
Yes, GPIO Viewer is perfect for the task. 

Thank you

**Reply by sfromis** [reply]
BTW. The correct approach is to always start the index at 1 when assigning gpios to component types, and not skip any numbers. The index is not meant as a number that has *any* more significance than to control the ordering.

**Reply by dali-dbr** [reply]
I understand, but for the task I need, it seems easier to have the same numbers as the actual GPIO inputs. Should I expect any issues by skipping numbers?

Thank you

**Reply by barbudor** [reply]
When not following the recommend rules, anything can happen
It may work for now until you add something else in the configuration or until something change in a future version.

You can use the recommend index starting at 1 and use the `Switchtext` command to set a specific name to each switch such as `GPIOx`

**Reply by sfromis** [reply]
It is not relevant what "seems easier" when it is not how Tasmota works. And the suggestion about using `SwitchText` to name your input is a good plan, maybe with much better names than after the gpio number.

**Comment by dali-dbr** [comment]
It makes perfect sense. Thank you for the Switchtext suggestion. Much better option indeed.

Br

**Comment by dali-dbr** [comment]
By the way.  Is there a way to set a Switchmode for all inputs using just one command or do I need to do it one by one?

SwitchMode1 1
SwitchMode2 1
SwitchMode3 1
....

Thank you

**Reply by barbudor** [reply]
~~One by one, but you only to do it once as it is then saved in flash settings~~

**Reply by sfromis** [reply]
The [docs page](https://tasmota.github.io/docs/Commands/#switchmode) says that "Index 0 applies to **all** switches". I'd try that.

**Comment by dali-dbr** [comment]
Perfect, everything works!  

So when I back up the configuration and restore it on a brand new Esp32 all this (configure module parameters, SetOption 141 1, Switchmode0 1, SwitchText....) will be included, or should these steps be applied for every new device?

![image](https://github.com/arendst/Tasmota/assets/6317229/607cf69e-5ff1-416e-8271-aa543652a682)

I'm sorry for the basic questions. 

Thank you so much!

**Reply by sfromis** [reply]
Yes, but .... beware that restoring config *to another device* gets to be a mess if you do not make sure to avoid duplicate names. The tool decode-config can work in a "more granular" way.

**Reply by dali-dbr** [reply]
Thank you. 

Do you mean this tool?
https://github.com/tasmota/decode-config

I was thinking more like having a backup configuration, in case one ESP dies, I can quickly restore it to a new one.

Thank you

**Reply by barbudor** [reply]
Yes, restoring on a replacement device is fine
It's using a backup of device A as a starting point for device B that can become messy
